spot-client/src/spot-tv/ui/components/setup/select-media/speaker-preview.js

```diff
@@ -77,10 +85,26 @@ export default class SpeakerPreview extends React.PureComponent {
             device.label === this.props.label);
 
         if (!description) {
+            this.setState({
+                hasSetSinkId: false
+            });
+
             return;
         }
 
-        this._ref.current.setSinkId(description.deviceId);
+        this._ref.current.setSinkId(description.deviceId)
+            .then(() => {
+                this.setState({
+                    hasSetSinkId: true
```

Comment:
does this async action take a lot of time ? I think this will be throwing errors when this gets resolved after the component's been unmounted

Reply:
No it doesn't take a lot of time. Maybe I handle unmount cases in a separate PR, maybe done tomorrow (May 21) or the day after (May 22)?

Reply:
ok